### PR TITLE
Fixed #605 custom content-type setting not effected sometimes

### LINKED ISSUE #605: add_response_header may fail sometimes
### Bug Description

```python
from robyn import Robyn, jsonify

app = Robyn(__file__)


@app.get("/")
async def h(request):
    return jsonify({"chinese": "中文"})


app.add_response_header("content-type", "application/json; charset=UTF-8")
app.start()
```

when refresh chrome, the output display two results:

`{"chinese":"中文"}`
or
`{"chinese":"涓枃"}`

so, I check it in DevTools,  the `Content-Type` change sometimes

`application/json; charset=UTF-8`
or
`text/plain`

### Steps to Reproduce

_No response_

### Your operating system

Windows

### Your Python version (`python --version`)

3.8

### Your Robyn version

None

### Additional Info

INFO:robyn.logger:Robyn version: 0.37.0

Python 3.8.3

### PR DESCRIPTION
**Pull request by 18o**
**Description**

This PR fixes #605 

`global_response_headers` in `Server` is a DashMap, which is case sensitive. 
The default key is `Content-Type` can be stay together with `content-type` in `global_response_headers`, so when it's extended , both `Content-Type` and `content-type` are written to http header without fixed sequence, but the browser is case insensitive, so it will take the last value in `Content-Type` and `content-type`, that's why this bug happened.

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -262,8 +262,16 @@ impl Server {
     /// Adds a new response header to our concurrent hashmap
     /// this can be called after the server has started.
     pub fn add_response_header(&self, key: &str, value: &str) {
-        self.global_response_headers
-            .insert(key.to_string(), value.to_string());
+        match key.to_lowercase().as_str() {
```
Comment (sansyrox) [inline src/server.rs:265]: Hey @18o 👋 

Thank you for the PR 😄 TIL about the web headers fact. Thank you ✨ 

However, I have one question. Why not make everything lowercase?

Comment (18o) [reply src/server.rs:265]: It's a good question, I was only thought about the `content-type` thing.
Do you mean all keys in the `global_response_headers` make them into lowercase?

Comment (sansyrox) [reply src/server.rs:265]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

> An HTTP header consists of its case-insensitive name followed by a colon (:), then by its value. [Whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) before the value is ignored.

I read this at MDN docs today.

Comment (sansyrox) [reply src/server.rs:265]: @18o , 
> Do you mean all keys in the global_response_headers make them into lowercase?

Comment (18o) [reply src/server.rs:265]: Yeah, we can make the keys in `global_response_headers ` all uppercase or lowercase, in order to avoid issues like this.

Comment (sansyrox) [reply src/server.rs:265]: @18o , sounds good. Let us make everything `lower_case` . I haven't seen someone mistype all upper case headers.

Comment (18o) [reply src/server.rs]: ok, it's implemented in the latest commit

**Comment by sansyrox** [comment]
hey @18o 👋 

Apologies for the delayed review here. I am working on a PR here - https://github.com/sparckles/Robyn/pull/704 , which might solve the issue at a more fundamental level.  I will merge it and hopefully, the issue should be fixed.

**Comment by sansyrox** [comment]
Hey @18o 👋 

Thank you for the PR. However, there was a more fundamental flaw in the code. It has been fixed in v0.47.0 😄

**Pull request closed without merging**